sympy/series/sequences.py

```diff
@@ -0,0 +1,522 @@
+
+    if step in [S.NegativeInfinity, S.Infinity]:
+        raise ValueError("step cannot be unbounded")
+
+    return (Interval(start, stop), sympify(step))
+
+
+class SeqExpr(SeqBase):
+    """Sequence expression class
+    Various sequences (SeqPer, SeqFormula, SeqFunc...) should inherit from
+    this class
+
+    Examples
+    ========
+
+    >>> from sympy.series.sequences import SeqExpr
+    >>> s = SeqExpr((1, 2, 3), (0, 10))
+    >>> s.gen
+    (1, 2, 3)
+    >>> s.interval
+    [0, 10]
+    >>> s.length
+    11
+
+    changing the step size
+
+    >>> SeqExpr((1, 2, 3), (0, 10, 2)).length
+    6
+
+    """
+
+    is_iterable = True
+
+    def __new__(cls, gen, interval=(0, S.Infinity, 1)):
+        bounds, step = _parse_interval(interval)
+        if bounds is S.EmptySet:
+            return S.EmptySequence
+        gen = sympify(gen)
+        interval = Tuple(bounds, step)
+        return Expr.__new__(cls, gen, interval)
+
+    def __iter__(self):
+        i = 0
+        while(i < self.length):
+            pt = self._ith_point(i)
+            yield self.coeff(pt)
+            i += 1
+
+    @property
+    def free_symbols(self):
+        """
+        This method returns the symbols in the object, excluding those
+        that take on a specific value (i.e. the dummy symbols).
+
+        Examples
+        ========
+
+        >>> from sympy import SeqFormula
+        >>> from sympy.abc import n, m
+        >>> SeqFormula((m*n**2, n), (0, 5)).free_symbols
+        set([m])
+        """
+        fsyms = set().union(*[a.free_symbols for a in self.args])
+        for d in self.dummy:
+            if d in fsyms:
+                fsyms.remove(d)
+        return fsyms
+
+    @property
+    def _gen(self):
+        return self.args[0]
+
+    @property
+    def _interval(self):
+        return self.args[1][0]
+
+    @property
+    def _start(self):
+        return self.interval.inf
+
+    @property
+    def _stop(self):
+        return self.interval.sup
+
+    @property
+    def _step(self):
+        return self.args[1][1]
+
+    @property
+    def _length(self):
+        return ceiling((self.stop - self.start + 1) / self.step)
+
+
+class SeqPer(SeqExpr):
+    """Represents a periodical sequence
+
+    The elements are repeated after a given period.
+
+    Examples
+    ========
+
+    >>> from sympy import SeqPer, oo
+    >>> s = SeqPer((1, 2, 3), (0, 5))
+    >>> s.periodical
+    (1, 2, 3)
+    >>> s.period
+    3
+
+    For value at a particular point
+
+    >>> s.coeff(3)
+    1
+
+    supports slicing
+
+    >>> s[:]
+    [1, 2, 3, 1, 2, 3]
+
+    iterable
+
+    >>> list(s)
+    [1, 2, 3, 1, 2, 3]
+
+    changing step size
+
+    >>> SeqPer((1, 2, 3), (0, 5, 2))[:]
+    [1, 3, 2]
+
+    sequence starts from negative infinity
+
+    >>> SeqPer((1, 2, 3), (-oo, 0))[0:6]
+    [1, 2, 3, 1, 2, 3]
+
+    See Also
+    ========
+
+    sympy.series.sequences.SeqFormula
+    sympy.series.sequences.SeqFunc
+    """
```

Comment:
I say we should allow the first one. I agree all values will not be reached. But then there is need to be that strict. Also, this leniency maybe useful when performing operations.

```
>>> SeqPer((0, 1, 2, 3), (0, 3)) + SeqPer((0, 1, 2, 3), (0, 2))
SeqPer((0, 1, 2, 3), (0, 2))
```
I would work on the validation part.